DESIGN.md

```diff
@@ -0,0 +1,66 @@
+1. Define the color or style:
+    * Define the color in `core/colors.xml`. Use good naming (`denim_blue`, `mine_shaft_gray` etc..). Our recommended tool for color naming is [Chirag Mehta - Name that color](http://chir.ag/projects/name-that-color/#6195ED).
+    * Define the style in `core/styles.xml`. For our default styles, applied with theme attributes, the naming convention is `Widget.KiwixTheme.*` and these should ideally inherit from their `Widget.MaterialComponent.*`, should no such parent exist, it is generally advisable to google "how to style [Component]". When there are two or more of a view that differ stylistically from our default styling, but are identical to each other visually, it is appropriate to extract a style to keep the XML DRY. A non-default style should use a descriptive name. e.x. `list_item_title`. 
+2. Add the color or style to a specific theme attribute in `themes.xml`.
+3. Make sure that the color or style works in both day and night mode. If it does not, add the dark mode compatible attribute to `values-night/themes.xml`.
+
+For a video on how to work with themes, styles and colors, see [Developing Themes with Style (Android Dev Summit '19)](https://www.youtube.com/watch?v=Owkf8DhAOSo).
+
+## Night Mode / Dark Theme
+Night mode is a different theme that has many benefits such as saving battery and improving visibility in low light conditions [2]. When doing any design, make sure the new design works in dark mode. Night mode can be activated from the systems settings or from the kiwix app. 
+
+Whenever a resource exists in both `res/*-night` and `res/*` it will be used as such. For example: `kiwix_icon_with_title.png` exists in both `res/drawable` and `res/drawable-night` which means that the image in `res/drawable-night` will be used automagically in night mode. For further reading on development with dark mode, see [Material Design - Dark Theme](https://developer.android.com/guide/topics/ui/look-and-feel/darktheme). 
+
+## Typography
+Text should contrast well, use the correct weight and size to present content as clearly and efficiently as possible. Google provides a "type scale" system that provides thirteen styles of typography to use for different contexts. Just as we do not give specific style attributes to views, we should not use specific text style attributes for text. Instead a type style should be used. For example, the button type style should be used for text on a button. To generate new type styles use the [Material Design type scale generator](https://material.io/design/typography/the-type-system.html#type-scale). To use type styles in practice, find a type style in `core/../res/type.xml` (e.x. `TextAppearance.KiwixTheme.Button`) and add it to a theme. Alternatively add a new type scale style to `core/../res/type.xml`. Additional reading can be found at [Material Design - Type System](https://material.io/design/typography/the-type-system.html#type-scale). 
```

Comment:
This isn't exactly right, so in `type.xml` we (should) customise the 13 types,
```
  <style name="TextAppearance.KiwixTheme.Headline1" parent="TextAppearance.MaterialComponents.Headline1">
    <!--Your custom type style here-->
  </style>
```
 we take our customised type and apply it in our theme so MaterialComponents will get these applied

```
<item name="textAppearanceHeadline1">@style/TextAppearance.KiwixTheme.Headline1</item>
```

And then it is perfectly fine on the view level to reference these attributes, see `item_help.xml`

```
  <TextView
    android:id="@+id/item_help_title"
    android:layout_width="0dp"
    android:layout_height="wrap_content"
    android:textAppearance="?textAppearanceHeadline6"
    app:layout_constraintEnd_toStartOf="@id/item_help_toggle_expand"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent"
    tools:text="@string/help_2" />
```
note the use of attribute `android:textAppearance="?textAppearanceHeadline6"` and not the type style.

Because a theme is not sufficient for the styling of a textview, as text varies so wildly in appearance across an app, the typography system was born to given 13 strict definitions of how text is displayed in Material Design

Reply:
Cool, thanks for clarifying.